File: exams-service/src/main/java/com/github/psinalberth/education/exam/adapter/persistence/QuestionEntity.java
package com.github.psinalberth.education.exam.adapter.persistence;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@Table(name = "questions", schema = "education")
public class QuestionEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String text;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "exam_id")
    private ExamEntity exam;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, mappedBy = "question")
    private List<QuestionOptionEntity> options;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "external_id")
    private String externalId;

    public void addOption(QuestionOptionEntity option) {
        if (this.options == null){
            this.options = new ArrayList<>();
        }
        option.setQuestion(this);
        this.options.add(option);
    }
}

File: questions-service/src/main/kotlin/com/github/psinalberth/education/question/adapter/persistence/QuestionEntity.kt
package com.github.psinalberth.education.question.adapter.persistence

import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.OneToMany
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity
@Table(name = "questions")
class QuestionEntity(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    val text: String,

    @OneToMany(cascade = [CascadeType.ALL], orphanRemoval = true, mappedBy = "question")
    var options: List<QuestionOptionEntity>? = null,

    @Column(name = "external_id")
    val externalId: String,

    @Column(name = "created_at")
    val createdAt: LocalDateTime,

    @Column(name = "updated_at")
    val updatedAt: LocalDateTime? = null
)